crypto/ec/ec_ameth.c

```diff
@@ -758,10 +758,11 @@ int ec_pkey_export_to(const EVP_PKEY *from, void *to_keydata,
     return rv;
 }
 
-static int ec_pkey_import_from(const OSSL_PARAM params[], void *key)
+static int ec_pkey_import_from(const OSSL_PARAM params[], void *vpctx)
 {
-    EVP_PKEY *pkey = key;
-    EC_KEY *ec = EC_KEY_new();
+    EVP_PKEY_CTX *pctx = vpctx;
+    EVP_PKEY *pkey = EVP_PKEY_CTX_get0_pkey(pctx);
+    EC_KEY *ec = EC_KEY_new_ex(pctx->libctx);
```

Comment:
So you create the low level key with the new constructor that takes a libctx...  but only for EC_KEY, not for the other key types.  Is that on purpose?  If it is, why do we have that sort of inconsistency?

Reply:
Because, so far, only EC_KEYs actual use that libctx during an import operation.

Reply:
... somewhere in the deep recesses of EC code, I assume?  Aren't we setting ourselves up for future mystery failures if we don't do this consistently across the board?

Reply:
Ok. I changed the others too - with the exception of ecx which currently doesn't have a libctx in the key structure(!)

Reply:
Good enough